File: 백준/Silver/3273. 두 수의 합/두 수의 합.java
// package baekjoon.solvedac.s3.q3273;

import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {


        Scanner scanner = new Scanner(System.in);
//        Array array = new Array();
//        long n = scanner.nextLong();
        int n = scanner.nextInt();
        long[]array = new long[(int)n];
        for (int i = 0; i < n; i++) {
            //long temp = scanner.nextLong();
//            array.push(temp);
            array[i] = scanner.nextLong();
        }
        long x = scanner.nextLong();

//        array.sort();
        Arrays.sort(array);
        int l=0;
        int r=n-1;
        long ret = 0;
        while(l<r){
            long sum = array[l]+array[r];
            if(sum==x){
                ret++;
                l++;r--;
            }else if(sum<x){
                l++;
            }else r--;
        }
        System.out.println(ret);

    }
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;
int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(0);cout.tie(0);
	
	ll n,x;
	cin>>n;
	vector<ll>v(n);
	for(int i=0;i<n;i++){
		cin>>v[i];
	}cin>>x;
	
	sort(v.begin(),v.end());
	
	ll l = 0, r = n-1;
	ll ret =0;
	
	while(l<r){
		ll sum = v[l]+v[r];
		if(sum==x){
			ret++;
			l++;
			r--;
		}else if(sum<x){
			l++;
		}else {
			r--;
		}
	}cout<<ret;
}